### DISCUSSION TITLE
Correct way to retrieve member models from declarative base?

### DISCUSSION THREAD

**Discussion by gzzo** [sqlalchemy/sqlalchemy#6239] [Usage Questions] [answered]
If I have a `Base = declarative_base()`, is there a proper way to get all the member models?

Previously I've used `Base.__subclasses__()`, but someone recently pointed out to me that this doesn't include inherited models when using joined table inheritance. 

We have a possible solution, `base._decl_class_registry.values()`, but the dictionary returned there has a `_sa_module_registry` object that makes iterating through it slightly more inelegant.

Really appreciate any help here!

**Comment by CaselIT** [comment]
Hi,

I believe the right approach would be to create a Base using [`registry.generate_base`](https://docs.sqlalchemy.org/en/14/orm/mapping_api.html?highlight=mappers#sqlalchemy.orm.registry.generate_base), then the list of models, that are connected to the mappers, can be accessed via [`registry.mappers`](https://docs.sqlalchemy.org/en/14/orm/mapping_api.html?highlight=mappers#sqlalchemy.orm.registry.mappers)

**Comment by zzzeek** [comment, marked as answer]
the "registry" in 1.4 should be there on the Base object too, I think...Base.registry.mappers

**Comment by zzzeek** [comment]
this is all SQLAlchemy 1.4 stuff.

**Comment by gzzo** [comment]
Thank you all for the help here. Is there a recommended approach when using v1.3? I'm still having trouble migrating to v1.4 due to typing issues when using mypy with stricter settings.
